src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/Utilities/LanguageServerTestComposition.cs

```diff
@@ -20,5 +23,26 @@ private static string GetDevKitExtensionPath()
         => Path.Combine(AppContext.BaseDirectory, DevKitExtensionSubdirectory, DevKitAssemblyFileName);
 
     public static Task<ExportProvider> CreateExportProviderAsync(ILoggerFactory loggerFactory, bool includeDevKitComponents)
-        => ExportProviderBuilder.CreateExportProviderAsync(extensionAssemblyPaths: [], devKitDependencyPath: includeDevKitComponents ? GetDevKitExtensionPath() : null, loggerFactory: loggerFactory);
+    {
+        var devKitDependencyPath = includeDevKitComponents ? GetDevKitExtensionPath() : null;
+        // Preload the devkit assemblies - the normal process hooks the resolving event, but that doesn't run at this entrypoint.
+        if (devKitDependencyPath != null)
+        {
+            foreach (var assembly in Directory.EnumerateFiles(Path.GetDirectoryName(devKitDependencyPath)!, "*.dll"))
+            {
+                AssemblyLoadContext.Default.LoadFromAssemblyPath(assembly);
+            }
+        }
```

Comment:
Oh hmm I was imagining we didn't really have tests that needed this but I guess we do? Then maybe it's best to share the hooking then?